esphome/components/eeh210/eeh210.cpp

```diff
@@ -0,0 +1,83 @@
+#include "eeh210.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace eeh210 {
+
+static const char *const TAG = "eeh210.sensor";
+
+static const uint8_t EEH210_I2C_COMMAND_TRIGGER_T_RH_MEASURE_HOLD = 0xE1;
+static const uint8_t EEH210_I2C_COMMAND_TRIGGER_T_MEASURE_HOLD = 0xE3;
+static const uint8_t EEH210_I2C_COMMAND_TRIGGER_RH_MEASURE_HOLD = 0xE5;
+
+static const uint8_t EEH210_I2C_COMMAND_TRIGGER_T_RH_MEASURE_NO_HOLD = 0xF1;
+static const uint8_t EEH210_I2C_COMMAND_TRIGGER_T_MEASURE_NO_HOLD = 0xF3;
+static const uint8_t EEH210_I2C_COMMAND_TRIGGER_RH_MEASURE_NO_HOLD = 0xF5;
+
+static const uint8_t EEH210_I2C_COMMAND_WRITE_USER_REGISTER = 0xE6;
+static const uint8_t EEH210_I2C_COMMAND_READ_USER_REGISTER = 0xFE;
+
+void EEH210Component::setup() { ESP_LOGCONFIG(TAG, "Setting up EEH210..."); }
+
+void EEH210Component::dump_config() {
+  ESP_LOGCONFIG(TAG, "EEH210:");
+  LOG_I2C_DEVICE(this);
+  switch (this->error_code_) {
+    case COMMUNICATION_FAILED:
+      ESP_LOGE(TAG, "Communication with EEH210 failed!");
+      break;
+    case CRC_CHECK_FAILED:
+      ESP_LOGE(TAG, "The crc check failed");
+      break;
+    case NONE:
+    default:
+      break;
+  }
+  LOG_UPDATE_INTERVAL(this);
+  LOG_SENSOR("  ", "Temperature", this->temperature_sensor_);
+  LOG_SENSOR("  ", "Humidity", this->humidity_sensor_);
+}
+
+float EEH210Component::get_setup_priority() const { return setup_priority::DATA; }
+void EEH210Component::update() {
+  uint8_t command[] = {EEH210_I2C_COMMAND_TRIGGER_T_RH_MEASURE_HOLD};
+  this->write(command, 1, true);
+  this->set_timeout(50, [this]() {
+    uint8_t i2c_response[6];
+    this->read(i2c_response, 6);
+
+    unsigned int raw_temperature = (i2c_response[1] << 8) | (i2c_response[2] & 0b11111100);
+    float temperature = -46.85 + 175.72 * (raw_temperature / 65536.0);
```

Comment:
We want floats, not doubles =)
```suggestion
    float temperature = -46.85f + 175.72f * (raw_temperature / 65536.0f);
```